f5/bigip/contexts.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+#
+# Copyright 2014 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+class TransactionContext(object):
+    def __init__(self, transaction):
+        self.transaction = transaction
+        self.bigip = transaction._meta_data['bigip']
+        self.icr = self.bigip._meta_data['icr_session']
+
+    def __enter__(self):
+        """Begins a new transaction context
+
+        When a transaction begins, this method will automatically be called
+        to set up the transaction.
+
+        Transaction IDs are automatically retrieved for you and the
+        appropriate headers are set so that future operations will reference
```

Comment:
We should probably say change "future operations" to something like: "operations in the Transaction Context", because it's not _all_ future operations, it's just the ones inside the "transaction".     This line of thinking makes me think we need to change the name of this class from:   "TransactionContext" to "TransactionContextManager".
What do you think?